### PR TITLE
Add platform/Span documentation.

### PR DESCRIPTION
**Pull request by pan-**
This is the associated documentation PR for ARMmbed/mbed-os#7828.

### CODE REVIEW

**Comment by pan-** [comment]
@donatieng Could you review ?

File: docs/reference/api/platform/Span.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+Span<uint8_t, 6> span_mac_address(mac_address);
+long_span = span_mac_address;
+```
+
+When you encode the size of the sequence viewed in the Span value, Span instances can view an empty sequence. The function `empty()` helps client code decide whether Span is viewing valid content or not.
+
+### Span class reference
+
+[![View code](https://www.mbed.com/embed/?type=library)](https://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/class_span.html)
+
```
Comment (AnotherButler) [inline docs/reference/api/platform/Span.md]: Please add an example.

File: docs/reference/api/platform/Span.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+```
+const uint8_t str[] = "Hello mbed!";
+
+Span<uint8_t> str_span(hello_mbed);
+
+ptrdiff_t half_size = str_span.size() / 2;
+
+Span<uint8_t> first_half = str_span.first(half_size);
+Span<uint8_t> second_half = str_span.last(half_size);
+Span<uint8_t> middle_half = str_span.subspan(/* offset */ half_size / 2, half_size);
```
Comment (donatieng) [inline docs/reference/api/platform/Span.md]: `middle_half` sounds a bit weird to me

**Comment by AnotherButler** [comment]
@pan-  Please address the comments from @donatieng

**Comment by ashok-rao** [comment]
@pan- Question: why do all code snippets show uint8_t ? why not other bit lengths? not supported yet or these are just examples?

**Comment by pan-** [comment]
@ashok-rao Any type is supported, it is just for examples purposes.

**Comment by pan-** [comment]
@donatieng I fixed your issue. 
@AnotherButler I've added an example and referenced the page in platform.

**Pull request merged**